File: Go/BruteForce/2798.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

var result int = 0
var M int = 0
var N int = 0

func blackJack(fin int, list []int, sum int, now int) {
	if fin == 3 {
		if M-sum <= M-result && M-sum >= 0 {
			result = sum
		}
		return
	} else {
		for i := now; i < N; i++ {
			blackJack(fin+1, list, sum+list[i], i+1)
		}
	}
}

func main() {
	reader := bufio.NewReader(os.Stdin)
	fmt.Fscanln(reader, &N, &M)
	list := make([]int, 0)
	input, _ := reader.ReadString('\n') //한줄 입력
	input_list := strings.Fields(input) //띄어쓰기로 구분
	for _, val := range input_list {
		temp, _ := strconv.Atoi(val)
		list = append(list, temp)
	}
	blackJack(0, list, 0, 0)
	fmt.Println(result)

}

File: python/BruteForce/2798.py
result=0
M=0
N=0
def BlackJack(fin:int,num:list,summ:int,now:int):
    global result
    global M
    global N
    if(fin==3):
        if(M-summ<=M-result and M-summ>=0): #반복문 시작이 모든 경우를 돌 수 있게, 0~5,1~5,2~5 이런식으로
            result=summ
        return

    else:
        for i in range(now,N):
            BlackJack(fin+1,num,summ+num[i],i+1)

    



if __name__ == "__main__":
    n_m_list=list(map(int,input().split(" ")))
    N=n_m_list[0]
    M=n_m_list[1]
    in_list=list(map(int,input().split()))
    BlackJack(0,in_list,0,0)

    print(result)
    
        


